public/app/features/alerting/unified/components/silences/SilenceTableRow.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { FC, Fragment, useState } from 'react';
+import { dateMath, GrafanaTheme, toDuration } from '@grafana/data';
+import { css } from '@emotion/css';
+import { Silence, SilenceMatcher } from 'app/plugins/datasource/alertmanager/types';
+import { AlertLabel } from '../AlertLabel';
+import { StateTag } from '../StateTag';
+import { CollapseToggle } from '../CollapseToggle';
+import { useRulesByMatcher } from '../../hooks/useRulesByMatcher';
+import { ActionButton } from '../rules/ActionButton';
+import { ActionIcon } from '../rules/ActionIcon';
+import { useStyles } from '@grafana/ui';
+import { RulesTable } from '../rules/RulesTable';
+
+interface Props {
+  className?: string;
+  silence: Silence;
+}
+
+const SilenceTableRow: FC<Props> = ({ silence, className }) => {
+  const [isCollapsed, setIsCollapsed] = useState<boolean>(true);
+
+  const styles = useStyles(getStyles);
+  const { status, matchers, startsAt, endsAt, comment, createdBy } = silence;
+  const matchingRules = useRulesByMatcher(matchers as SilenceMatcher[]);
+
+  const dateDisplayFormat = 'YYYY-MM-DD HH:mm';
+  const startsAtDate = dateMath.parse(startsAt);
+  const endsAtDate = dateMath.parse(endsAt);
+  const duration = toDuration(endsAtDate?.diff(startsAtDate || '')).humanize();
```

Comment:
This isn't the best method and deviates from the figma designs, but `moment` doesn't provide a format method and  is no longer a maintained library and they now recommend using a different library. There is a plugin for moment, `moment-duration-format` that we may be able to use in the interim and file an issue for updating the `moment_wrapper.ts` to use a different library such as `day.js`

Reply:
I'm not sure how the figma design looks but I think you should use the `dateTimeFormat` function. It will default to the default grafana date format and it will also use potential time zone overrides applied by the user in the Grafana instance.

Reply:
hmm, or you want do print the duration?

Reply:
It might be something that we should add support for in the grafana/data if it is missing.

Reply:
Thanks @mckn I'll look into that as I'd really like to not use moment

Reply:
For now I'm just displaying the duration in seconds and will have a duration format option in my next PR.

Reply:
Super, sounds great! Let me know if you need help figuring out how to add that to grafana.